modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -205,6 +206,13 @@ public CPDefinition addCPDefinition(
 
 		// Commerce product definition
 
+		if ((subscriptionLength < 1) || (maxSubscriptionCycles < 0) ||
+			(deliverySubscriptionLength < 1) ||
+			(deliveryMaxSubscriptionCycles < 0)) {
+
+			throw new CPSubscriptionDateException();
```

Comment:
@victorcdp , I want the lengths and subscriptionCycles to be separate checks with their own exception messages.

Ex.
```
			throw new CPDefinitionExpirationDateException(
				"Enter subscription length greater than or equal to 1");
```
```
			throw new CPDefinitionExpirationDateException(
				"Enter subscription cycles greater than or equal to 0");
```

Reply:
Should be good now. We have to run sf again right?